internal/auth.go

```diff
@@ -56,31 +56,51 @@ func ValidateCookie(r *http.Request, c *http.Cookie) (string, error) {
 }
 
 // Validate email
-func ValidateEmail(email string) bool {
+func ValidateEmail(email string, rule string) bool {
 	found := false
-	if len(config.Whitelist) > 0 {
-		for _, whitelist := range config.Whitelist {
-			if email == whitelist {
-				found = true
-			}
-		}
+
+	_, ruleExists := config.Rules[rule]
```

Comment:
To make the following code shorter, could you rename the function parameter to `ruleName` and then assign the first return argument of this function into a variable `rule`:
```go
rule, ruleExists := config.Rules[ruleName]
```